lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -0,0 +1,314 @@
+        execute_dependant(:refund, money, authorization, options)
+      end
+
+      # Public: Voids previous authorization
+      #
+      # authorization - Reference to previous authorization, obtained from response object returned by authorize
+      # options       - Hash containing capture options
+      #                 :order_id         Merchant created id for the authorization (optional)
+      #
+      # Returns Active Merchant response object
+      def void(authorization, options={})
+        execute_dependant(:void, nil, authorization, options)
+      end
+
+      # Public: Verifies credit card. Does this by doing a authorization of 1.00 Euro and then voiding it.
+      #
+      # credit_card - Credit card
+      # options     - Hash containing authorization options
+      #               :order_id         Merchant created id for the authorization
+      #               :billing_address  Hash with billing address information
+      #               :description      Merchant created authorization description (optional)
+      #               :currency         Sale currency to override money object or default (optional)
+      #               :given            Customer given name to override card holder's (optional)
+      #               :family           Customer family name to override card holder's (optional)
+      #
+      # Returns Active Merchant response object of Authorization operation
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      private
+
+      # Private: Execute purchase or authorize operation
+      def execute_new_order(action, money, credit_card, options)
+        request = build_request do |xml|
+          add_identification_new_order(xml, options)
+          add_payment(xml, action, money, options)
+          add_account(xml, credit_card)
+          add_customer(xml, credit_card, options)
+        end
+
+        commit(request)
+      end
+
+      # Private: Execute operation that depends on athorization code from previous purchase or authorize operation
+      def execute_dependant(action, money, authorization, options)
+        request = build_request do |xml|
+          add_identification_authorization(xml, authorization, options)
+          add_payment(xml, action, money, options)
+        end
+
+        commit(request)
+      end
+
+      # Private: Build XML wrapping code yielding to code to fill the transaction information
+      def build_request
+        builder = Nokogiri::XML::Builder.new(:encoding => 'UTF-8') do |xml|
+          xml.Request(:version => "1.0") do
+
+            xml.Header { xml.Security(:sender => @options[:sender_id]) }
+
+            xml.Transaction(:mode => test? ? 'CONNECTOR_TEST' : 'LIVE', :response => 'SYNC', :channel => @options[:channel_id]) do
```

Comment:
@girasquid The system is very flexible. There are two test modes plus live, we are only using the test mode that test the connector. Additionally, it is possible to do test requests to the live system, the only difference is that in the live system there is a transaction fee for test requests and in the test server they are for free. We do not want to do the test request to the live server, because if someone runs the remote test many times will cost money.